### DISCUSSION TITLE
How do I call pyinstallerl's built-in python interpreter

### DISCUSSION THREAD

**Discussion by Jayce-liang** [pyinstaller/pyinstaller#7863] [Help]
I want to use pyinstaller's built-in python interpreter, but it doesn't seem to work.

![image](https://github.com/pyinstaller/pyinstaller/assets/76100439/a4b5cc67-e0f8-4ff9-aec9-f90883e0e169)

**Comment by rokm** [comment]
That's because your executable (`sys.executable`) is not a python interpreter; so trying to run it like that will just run another instance of your program. You need to add a custom/special argument to let your program know that, instead of running your main application code, it should load the given .py file and `exec` it (and you will of course need to implement that, too). 

See #5978 for an example

**Reply by Jayce-liang** [reply]
thank you so much, i will try it.

**Reply by bwoodsend** [reply]
Note that, even unfrozen you should use `python_path = sys.executable`. Otherwise you may get a different Python environment without your packages installed or you might not get any Python environment at all.

**Reply by Jayce-liang** [reply]
Thanks for correcting

**Reply by Jayce-liang** [reply]
For some reason, I can't use exec()  https://github.com/orgs/pyinstaller/discussions/5978, I need to use subprocess. Popen()。 So there seems to be no way to do it now.

**Reply by bwoodsend** [reply]
> For some reason, I can't use exec()

Python is calling `exec()` under the hood all over the place so I'd be very surprised if that's really true. What goes wrong when you use `exec()`?

**Reply by Jayce-liang** [reply]
At first I thought using exec() will blocks the main thread and can not get the real-time output from the exec() task. And there doesn't seem to be a good way to force the exec() task to terminate.

--- but now it seems ok😂
![image](https://github.com/pyinstaller/pyinstaller/assets/76100439/134d5a1c-20f1-42a2-86be-b7fdb7c19786)



I now create a static virtual environment and place it in the project directory. Now using the Python interpreter for the virtual environment, this seems to work.🤣
